### PR TITLE
Feature/mico207 DELETE /applications/{shortName}/{version}

### LINKED ISSUE #207: Implement DELETE /applications/{shortName}/{version}
As a developer I want to delete an application.

### PR DESCRIPTION
**Pull request by Pascal157600**
- [ ] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Ensure to use auto format in **all** files
- [ ] Tests created for changes or:
  - [ ] Test cases are missing - opened issue for that:
  - [ ] Reason for missing tests:
- [ ] Screenshots added (for UI changes)
- [ ] Change in CHANGELOG.md described

---

## Resolving Issue/ Feature
resolves #207

### CODE REVIEW

File: mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java [hunk 1]
```diff
@@ -63,17 +63,36 @@
...
     }
 
+    @DeleteMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}")
+    public ResponseEntity deleteApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName,
+                                            @PathVariable(PATH_VARIABLE_VERSION) String version) {
+        Optional<Application> applicationOptional = applicationRepository.findByShortNameAndVersion(shortName, version);
+        if (!applicationOptional.isPresent()) {
+            return ResponseEntity.notFound().build();
+        } else {
+            //only allow a deletion if the service has no dependees or dependers
```
Comment (JanVetter) [inline mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java]: I would differentiate between `dependees` and `dependers`.

File: mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java [hunk 2]
```diff
@@ -63,17 +63,36 @@
...
+            Application application = applicationOptional.get();
+            //TODO: && getDependers(shortName,version) == null --> check for dependers missing
+            if (application.getDependsOn() == null) {
```
Comment (JanVetter) [inline mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java]: Why does the `DependsOn` have to be `null`?

File: mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java [hunk 3]
```diff
@@ -63,17 +63,36 @@
...
+                applicationRepository.deleteApplicationByShortNameAndVersion(shortName, version);
+                return ResponseEntity.ok().build();
+            } else {
+                return ResponseEntity.badRequest().build();
```
Comment (JanVetter) [inline mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java]: We might simplify the code structure..
```
Optional<Application> applicationOptional = applicationRepository.findByShortNameAndVersion(shortName, version);
if (!applicationOptional.isPresent()) {
    return ResponseEntity.notFound().build();
}

//only allow a deletion if the service has no dependees or dependers
Application application = applicationOptional.get();
//TODO: && getDependers(shortName,version) == null --> check for dependers missing
if (application.getDependsOn() == null) {
    applicationRepository.deleteApplicationByShortNameAndVersion(shortName, version);
    return ResponseEntity.ok().build();
}

return ResponseEntity.badRequest().build();
```
What do you think?

File: mico-core/src/test/java/io/github/ust/mico/core/ApplicationControllerTest.java [hunk 4]
```diff
@@ -93,11 +93,24 @@ public void createApplication() throws Exception {
...
         result.andExpect(status().isCreated());
     }
 
+    @Test
+    public void deleteApplication() throws Exception {
+        Application application = new Application(SHORT_NAME, VERSION, DESCRIPTION);
+
+        given(applicationRepository.findByShortNameAndVersion(SHORT_NAME, VERSION)).willReturn(Optional.of(application));
+
+        ResultActions resultDelete = mvc.perform(delete(BASE_PATH + SHORT_NAME + "/" + VERSION)
```
Comment (JanVetter) [inline mico-core/src/test/java/io/github/ust/mico/core/ApplicationControllerTest.java]: We might think of introducing a new variable for `BASE_PATH + SHORT_NAME + "/" + VERSION`.

**Comment by st116290** [comment]
Superseded by #347

**Pull request closed without merging**